MetadataExtractor/Formats/QuickTime/QuickTimeTiffHandler.cs

```diff
@@ -22,7 +24,11 @@ public override void SetTiffMarker(int marker)
             {
                 throw new TiffProcessingException($"Unexpected TIFF marker: 0x{marker:X}");
             }
-            PushDirectory(new T());
+            var directory = new T
+            {
+                Parent = Directories.OfType<TParent>().FirstOrDefault()
```

Comment:
This is prone to selecting the wrong parent. The first one might not be the correct one. I would suggest adding the parent directory as a constructor parameter, storing it in a field, then adding it here.

You can then remove the `TParent` type parameter.

Reply:
You mean an additional directory stack in `QucikTimeMetadataReader`? I'm not sure if adding state to the latter is better than the current solution.

Reply:
Isn't `SetTiffMarker` only called once? I don't think a stack is needed.

Reply:
I meant the state would be in this class (`QuickTimeTiffHandler`).

Reply:
The reader encounters the atoms in order:

* CMT1
* CMT2
* CMT3
* CMT4

For each of those, a new `ITiffHandler` instance is created, meaning a directory stack created from scratch every time.

Reply:
Thanks for clarifying. I still feel it should be possible to pass the parent forward somehow rather than looking it up, but will have to pull down your branch to play around with it.